### ISSUE TITLE
go get failed

### ISSUE THREAD

**Issue by likezjuisee** [3d0c/gmf#68]
root@ubuntu-HP-ProDesk-680-G2-MT:~/software/ffmpeg-3.4.2# go get github.com/3d0c/gmf
# github.com/3d0c/gmf
../../go/src/github.com/3d0c/gmf/packet.go: In function ‘_cgo_5625a69e2c98_Cfunc_av_copy_packet’:
../../go/src/github.com/3d0c/gmf/packet.go:52:2: warning: ‘av_copy_packet’ is deprecated [-Wdeprecated-declarations]
 func (p *Packet) Decode(cc *CodecCtx) (*Frame, bool, int, error) {
  ^
In file included from ../../go/src/github.com/3d0c/gmf/packet.go:7:0:
/usr/local/include/libavcodec/avcodec.h:4630:5: note: declared here
 int av_copy_packet(AVPacket *dst, const AVPacket *src);
     ^
../../go/src/github.com/3d0c/gmf/packet.go: In function ‘_cgo_5625a69e2c98_Cfunc_avcodec_decode_audio4’:
../../go/src/github.com/3d0c/gmf/packet.go:87:2: warning: ‘avcodec_decode_audio4’ is deprecated [-Wdeprecated-declarations]
In file included from ../../go/src/github.com/3d0c/gmf/packet.go:7:0:
/usr/local/include/libavcodec/avcodec.h:4944:5: note: declared here
 int avcodec_decode_audio4(AVCodecContext *avctx, AVFrame *frame,
     ^
../../go/src/github.com/3d0c/gmf/packet.go: In function ‘_cgo_5625a69e2c98_Cfunc_avcodec_decode_video2’:
../../go/src/github.com/3d0c/gmf/packet.go:104:2: warning: ‘avcodec_decode_video2’ is deprecated [-Wdeprecated-declarations]
   if ret < 0 {
  ^
In file included from ../../go/src/github.com/3d0c/gmf/packet.go:7:0:
/usr/local/include/libavcodec/avcodec.h:4993:5: note: declared here
 int avcodec_decode_video2(AVCodecContext *avctx, AVFrame *picture,
     ^
# github.com/3d0c/gmf
../../go/src/github.com/3d0c/gmf/stream.go: In function ‘_cgo_5625a69e2c98_Cfunc_avcodec_copy_context’:
../../go/src/github.com/3d0c/gmf/stream.go:45:2: warning: ‘avcodec_copy_context’ is deprecated [-Wdeprecated-declarations]
  // and it should be decoder.
  ^
In file included from /usr/local/include/libavformat/avformat.h:319:0,
                 from ../../go/src/github.com/3d0c/gmf/stream.go:7:
/usr/local/include/libavcodec/avcodec.h:4412:5: note: declared here
 int avcodec_copy_context(AVCodecContext *dest, const AVCodecContext *src);
     ^
# github.com/3d0c/gmf
/usr/local/lib/libavutil.a(hwcontext_vaapi.o): In function `vaapi_device_create':
/home/ubuntu/software/ffmpeg-3.4.2/libavutil/hwcontext_vaapi.c:1197: undefined reference to `vaGetDisplay'
/home/ubuntu/software/ffmpeg-3.4.2/libavutil/hwcontext_vaapi.c:1221: undefined reference to `vaGetDisplayDRM'
/usr/local/lib/libavutil.a(hwcontext_vdpau.o): In function `vdpau_device_create':
/home/ubuntu/software/ffmpeg-3.4.2/libavutil/hwcontext_vdpau.c:439: undefined reference to `vdp_device_create_x11'
collect2: error: ld returned 1 exit status

root@ubuntu-HP-ProDesk-680-G2-MT:~/software/ffmpeg-3.4.2# ffmpeg -version
ffmpeg version 3.4.2 Copyright (c) 2000-2018 the FFmpeg developers
built with gcc 5.4.0 (Ubuntu 5.4.0-6ubuntu1~16.04.9) 20160609
configuration: --prefix=/usr/local
libavutil      55. 78.100 / 55. 78.100
libavcodec     57.107.100 / 57.107.100
libavformat    57. 83.100 / 57. 83.100
libavdevice    57. 10.100 / 57. 10.100
libavfilter     6.107.100 /  6.107.100
libswscale      4.  8.100 /  4.  8.100
libswresample   2.  9.100 /  2.  9.100

**Comment by 3d0c** [comment]
Yes. New api is unfinished for now. I have pushed last changes, so it should build, but most of examples and tests won't work.

**Comment by abrougbacem** [comment]
any solution please
